pandas/tools/merge.py

```diff
@@ -193,6 +195,13 @@ def __init__(self, left, right, how='inner', on=None,
                 'can not merge DataFrame with instance of '
                 'type {0}'.format(type(right)))
 
+        # warn user when merging between different levels
+        if left.columns.nlevels != right.columns.nlevels:
```

Comment:
ahh didn't even notice, this is ONLY checking the ``levels``, it *should* be doing something like:

```
if left._get_axis(axis).nlevels != right._get_axis(axis).nlevels:
   ...
```